torch/csrc/jit/script/compiler.cpp

```diff
@@ -166,7 +173,9 @@ struct Environment {
 
     if (!retval && (retval = findInParentFrame(ident)) &&
         getBlockOwningKind() == prim::Loop) {
-      retval = createCapturedInput(ident);
+      if(Value* simple_val = asSimple(retval)) {
+        retval = createCapturedInput(simple_val, ident);
+      }
```

Comment:
Tuple values are first-class in the graph, so they are SimpleValues and are captured. This was a bug in the last commit: we shouldn't try to capture other SugaredValues. The reason we create a captured input in getVar is because we are concerned that a future setVar will cause it to be come a loop-carried dependency. For SugaredValues, this case is prevented in setVar (you cannot reassign a sugared value).